### DISCUSSION TITLE
Migrating from PECL/Stats to Packagist/MathPHP

### DISCUSSION THREAD

**Discussion by ghost** [markrogoyski/math-php#480] [Q&A]
Hi Team,

I am currently using below functions from PECL/Stats extension. I have enabled the extension on Azure Web App.

            1. stats_cdf_normal
            2. stats_dens_normal

PECL/Stats does not have a .dll for PHP8. So I am trying to switch to MathPHP
Could you please let me know what are the corresponding functions in MathPHP for above 2 functions ?
Also could you please tell me if MathPHP has dll file for PHP 8.2 ?   

Kind Regards
Tharanga

**Comment by markrogoyski** [comment]
Hi Tharanga,

Thank you for your interest in MathPHP.

MathPHP works with PHP 8.2. There are no system libraries or modification needed to PHP. You can install MathPHP as a standard PHP composer package. Instructions are on the [main page](https://github.com/markrogoyski/math-php?tab=readme-ov-file#setup).

For similar stats functions, you can check out the [MathPHP\Probability\Distribution\Continuous](https://github.com/markrogoyski/math-php?tab=readme-ov-file#probability---continuous-distributions) namespace in MathPHP. There are no global functions, but rather individual classes that provide similar functionality. These functions may be what you are looking for:

```php
use MathPHP\Probability\Distribution\Continuous;

$σ = 1;
$μ = 0;
$x = 2;

$normal = new Continuous\Normal($μ, $σ);
$pdf    = $normal->pdf($x);
$cdf    = $normal->cdf($x);
```

**Reply by ghost** [reply]
Thank you so much , i wrote some test code and seems all working. Really appreciate your response/
